api-put-object-common.go

```diff
@@ -77,7 +78,7 @@ func optimalPartInfo(objectSize int64) (totalPartsCount int, partSize int64, las
 
 // getUploadID - fetch upload id if already present for an object name
 // or initiate a new request to fetch a new upload id.
-func (c Client) newUploadID(bucketName, objectName string, metaData map[string][]string) (uploadID string, err error) {
+func (c Client) newUploadID(ctx context.Context, bucketName, objectName string, opts *PutObjectOptions) (uploadID string, err error) {
```

Comment:
I like PutObjectOptions change, but couldn't we do it in a separate PR ? (let's keep it as it now since I am already late reviewing this PR)

Reply:
@vadmeste, the PutObjectOptions change is the main reason for this PR. Context change is minor since we are just passing it through to the executeMethod everywhere so that request can get context. It made sense to combine the changes in one PR.